nat-lab/tests/test_direct_connection.py

```diff
@@ -269,6 +269,122 @@ async def test_direct_working_paths(
         async with Ping(alpha_connection, beta.ip_addresses[0]).run() as ping:
             await testing.wait_long(ping.wait_for_next_ping())
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize("setup_params, reflexive_ip", UHP_WORKING_PATHS)
+async def test_direct_working_paths_are_reestablished(
+    setup_params: List[SetupParameters],
+    reflexive_ip: str,
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await setup_mesh_nodes(exit_stack, setup_params)
+        alpha, beta = env.nodes
+        alpha_client, beta_client = env.clients
+        alpha_connection, _ = [conn.connection for conn in env.connections]
+
+        await testing.wait_defined(
+            asyncio.gather(
+                alpha_client.wait_for_state_peer(
```

Comment:
State check not needed here, if both of nodes have direct feature enabled, state is check in `setup_mesh_nodes`